src/PufferVaultMainnet.sol

```diff
@@ -7,61 +7,110 @@ import { ILidoWithdrawalQueue } from "src/interface/Lido/ILidoWithdrawalQueue.so
 import { IEigenLayer } from "src/interface/EigenLayer/IEigenLayer.sol";
 import { IStrategy } from "src/interface/EigenLayer/IStrategy.sol";
 import { IWETH } from "src/interface/Other/IWETH.sol";
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
 
 /**
  * @title PufferVault
  * @author Puffer Finance
  * @custom:security-contact security@puffer.fi
  */
 contract PufferVaultMainnet is PufferVault {
-    // slither-disable-next-line unused-state
-    IWETH internal constant _WETH = IWETH(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+    error ETHTransferFailed();
+
+    /**
+     * Emitted when the daily withdrawal limit is set
+     * @dev Signature: 0x8d5f7487ce1fd25059bd15204a55ea2c293160362b849a6f9244aec7d5a3700b
+     */
+    event DailyWithdrawalLimitSet(uint96 oldLimit, uint96 newLimit);
+
+    /**
+     * @dev The Wrapped Ethereum ERC20 token
+     */
+    IWETH internal immutable _WETH;
 
     constructor(
         IStETH stETH,
+        IWETH weth,
         ILidoWithdrawalQueue lidoWithdrawalQueue,
         IStrategy stETHStrategy,
         IEigenLayer eigenStrategyManager
     ) PufferVault(stETH, lidoWithdrawalQueue, stETHStrategy, eigenStrategyManager) {
-        _ST_ETH = stETH;
-        _LIDO_WITHDRAWAL_QUEUE = lidoWithdrawalQueue;
-        _EIGEN_STETH_STRATEGY = stETHStrategy;
-        _EIGEN_STRATEGY_MANAGER = eigenStrategyManager;
-        // Approve stETH to Lido && EL
+        _WETH = weth;
         _disableInitializers();
     }
 
     /**
      * @notice Changes token from stETH to WETH
      */
     function initialize() public reinitializer(2) {
-        ERC4626Storage storage $ = _getERC4626StorageInternal();
-        $._asset = _WETH;
+        // In this initialization, we swap out the underlying stETH with WETH
+        ERC4626Storage storage erc4626Storage = _getERC4626StorageInternal();
+        erc4626Storage._asset = _WETH;
+
+        VaultStorage storage $ = _getPufferVaultStorage();
+        $.lastWithdrawalDay = uint64(block.timestamp / 1 days);
```

Comment:
The daily limit needs to be set to 0, as in withdrawal is disabled. 
(We need to explicitly set that to 0 here for readability and auditors.)